fiware-region-sanity-tests/README.rst

```diff
@@ -225,6 +236,74 @@ provided as part of this component, which has to be installed together with the
 rest of standard parsers bundled in NGSI Adapter package.
 
 
+DBus and HTTP PhoneHome Service for E2E tests
+---------------------------------------------
+
+Some E2E test cases have been implemented to check the connection in both
+*Internet -> VM* and *VM -> Internet*.
+This test cases are:
+
+* Test whether it is possible to deploy and instance, assign an allocated
+  public IP and establish a SSH connection *(Internet -> VM)*
+* Test whether it is possible to deploy an instance
+  and connect to INTERNET (SNAT) without assigning a public IP *(VM -> Internet)*
+
+The later will try to execute a *PhoneHome request* (executed by Cloud-Init in the VM)
+to the *HTTP PhoneHome service* running in the configured HOST:PORT
+(*phonehome_endpoint* configuration). If this value is not set, this test will be skipped.
+
+The test uses two components:
+
+- A HTTP/Dbus PhoneHome server, that is launched as a service in the same host where test is executed (with public IP).
+- A DBus client used by test implementation to wait for PhoneHome requests through the HTTP PhoneHome server.
+
+The implemented PhoneHome service uses the DBus system technology to communicate the
+test execution and the HTTP PhoneHome server who receives the PhoneHome request from
+deployed VMs.
+
+
+**HTTP PhoneHome server**
+
+The HTTP PhoneHome server waits for *POST HTTP requests* from VMs.
+This service publishes a DBus object (DBus server) to be used by tests to wait for
+PhoneHome requests.
+
+When any request is received, HTTP PhoneHome server will emit over the published object a
+signal to inform to all connected tests about the event (broadcasting). This signal contains the
+hostname of the VM (the one received in the HTTP POST body). This signal will be take in account by
+tests that waits for a signal with this hostname value; rest of tests will ignore it and they keep on
+listening new signals with the correct data to them.
+
```

Comment:
When a request is received, HTTP PhoneHome server will response, through the published object, a
+signal to inform all connected tests about the event (broadcasting). This signal contains the
+hostname of the VM (the one received in the HTTP POST body). This signal will be take into account by
+tests that are waiting for a signal with this hostname value; the other tests will ignore it and will keep on
+listening new signals with the correct data (i.e. correct hostname) to them.